Compile failed for Spark 3.2.0

**Describe the bug**

```text
[ERROR] [Error] /blaze/spark-extension-shims-spark3/src/main/scala/org/apache/spark/sql/blaze/InterceptedValidateSparkPlan.scala:84: object InvalidAQEPlanException is not a member of package org.apache.spark.sql.execution.adaptive
```

**To Reproduce**
```bash
mvn clean package -DskipTests -Prelease -Pspark-3.2 -DsparkVersion=3.2.0
```


**Expected behavior**
Compilation successful.

--- a/spark-extension-shims-spark3/src/main/java/org/apache/spark/sql/blaze/ValidateSparkPlanInjector.java
+++ b/spark-extension-shims-spark3/src/main/java/org/apache/spark/sql/blaze/ValidateSparkPlanInjector.java
@@ -24,18 +24,27 @@ import net.bytebuddy.dynamic.ClassFileLocator;
 import net.bytebuddy.dynamic.loading.ClassLoadingStrategy;
 import net.bytebuddy.implementation.MethodDelegation;
 import net.bytebuddy.pool.TypePool;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
 
 public class ValidateSparkPlanInjector {
+
+    private static final Logger logger = LoggerFactory.getLogger(ValidateSparkPlanInjector.class);
+
     public static void inject() {
-        ByteBuddyAgent.install();
-        TypeDescription typeDescription = TypePool.Default.ofSystemLoader()
-                .describe("org.apache.spark.sql.execution.adaptive.ValidateSparkPlan$")
-                .resolve();
-        new ByteBuddy()
-                .redefine(typeDescription, ClassFileLocator.ForClassLoader.ofSystemLoader())
-                .method(named("apply"))
-                .intercept(MethodDelegation.to(ValidateSparkPlanApplyInterceptor.class))
-                .make()
-                .load(ClassLoader.getSystemClassLoader(), ClassLoadingStrategy.Default.INJECTION);
+        try {
+            ByteBuddyAgent.install();
+            TypeDescription typeDescription = TypePool.Default.ofSystemLoader()
+                    .describe("org.apache.spark.sql.execution.adaptive.ValidateSparkPlan$")
+                    .resolve();
+            new ByteBuddy()
+                    .redefine(typeDescription, ClassFileLocator.ForClassLoader.ofSystemLoader())
+                    .method(named("apply"))
+                    .intercept(MethodDelegation.to(ValidateSparkPlanApplyInterceptor.class))
+                    .make()
+                    .load(ClassLoader.getSystemClassLoader(), ClassLoadingStrategy.Default.INJECTION);
+        } catch (TypePool.Resolution.NoSuchTypeException e) {
+            logger.warn("No such type of ValidateSparkPlan", e);
+        }
     }
 }

--- a/spark-extension/src/main/scala/org/apache/spark/sql/blaze/BlazeConverters.scala
+++ b/spark-extension/src/main/scala/org/apache/spark/sql/blaze/BlazeConverters.scala
@@ -597,11 +597,30 @@ object BlazeConverters extends Logging {
     // for enabling filter-project optimization in native side
     getPartialAggProjection(exec.aggregateExpressions, exec.groupingExpressions) match {
       case Some((transformedAggregateExprs, transformedGroupingExprs, projections)) =>
-        return convertHashAggregateExec(
-          exec.copy(
-            aggregateExpressions = transformedAggregateExprs,
-            groupingExpressions = transformedGroupingExprs,
-            child = convertProjectExec(ProjectExec(projections, exec.child))))
+        val transformedExec =
+          try {
+            exec.copy(
+              aggregateExpressions = transformedAggregateExprs,
+              groupingExpressions = transformedGroupingExprs,
+              child = convertProjectExec(ProjectExec(projections, exec.child)))
+          } catch {
+            case _: NoSuchMethodError =>
+              import scala.reflect.runtime.universe._
+              import scala.reflect.runtime.currentMirror
+              val mirror = currentMirror.reflect(exec)
+              val copyMethod = typeOf[HashAggregateExec].decl(TermName("copy")).asMethod
+              val params = copyMethod.paramLists.flatten
+              val args = params.map { param =>
+                param.name.toString match {
+                  case "aggregateExpressions" => transformedAggregateExprs
+                  case "groupingExpressions" => transformedGroupingExprs
+                  case "child" => convertProjectExec(ProjectExec(projections, exec.child))
+                  case _ => mirror.reflectField(param.asTerm).get
+                }
+              }
+              mirror.reflectMethod(copyMethod)(args: _*).asInstanceOf[HashAggregateExec]
+          }
+        return convertHashAggregateExec(transformedExec)
       case None => // passthrough
     }
 

